static/js/reducers/ui.js

```diff
@@ -0,0 +1,56 @@
+import {
+  CLEAR_UI,
+  UPDATE_DIALOG_TEXT,
+  UPDATE_DIALOG_TITLE,
+  SET_DIALOG_VISIBILITY,
+  SET_WORK_HISTORY_EDIT,
+  SET_WORK_DIALOG_VISIBILITY,
+  SET_WORK_DIALOG_INDEX,
+} from '../actions/ui';
+
+const INITIAL_UI_STATE = {};
+
+export const ui = (state = INITIAL_UI_STATE, action) => {
+  switch (action.type) {
+  case UPDATE_DIALOG_TEXT:
+    return Object.assign({}, state, {
+      dialog: Object.assign(
+        {},
+        state.dialog,
+        { text: action.payload }
+      )
+    });
+  case UPDATE_DIALOG_TITLE:
+    return Object.assign({}, state, {
+      dialog: Object.assign(
+        {},
+        state.dialog,
+        { title: action.payload }
+      )
+    });
+  case SET_DIALOG_VISIBILITY:
+    return Object.assign({}, state, {
+      dialog: Object.assign(
+        {},
+        state.dialog,
+        { visible: action.payload }
+      )
+    });
+  case SET_WORK_HISTORY_EDIT:
+    return Object.assign({}, state, {
+      workHistoryEdit: action.payload
```

Comment:
Since this is a boolean, can you have this default to `false` in `INITIAL_UI_STATE`? This will also avoid the need for checking for `undefined` since it will always be true or false